engine/src/test/java/io/camunda/zeebe/engine/processing/incident/ServiceTaskLikeIncidentTest.java

```diff
@@ -11,46 +11,59 @@
 import static org.assertj.core.api.Assertions.entry;
 
 import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.engine.util.ServiceTaskLikeBuilderProvider;
+import io.camunda.zeebe.engine.util.TaskBuilder;
 import io.camunda.zeebe.model.bpmn.Bpmn;
 import io.camunda.zeebe.model.bpmn.BpmnModelInstance;
-import io.camunda.zeebe.model.bpmn.builder.ServiceTaskBuilder;
+import io.camunda.zeebe.model.bpmn.builder.AbstractServiceTaskLikeBuilder;
 import io.camunda.zeebe.protocol.record.Assertions;
 import io.camunda.zeebe.protocol.record.Record;
 import io.camunda.zeebe.protocol.record.intent.IncidentIntent;
 import io.camunda.zeebe.protocol.record.intent.JobIntent;
 import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
-import io.camunda.zeebe.protocol.record.value.BpmnElementType;
 import io.camunda.zeebe.protocol.record.value.ErrorType;
-import io.camunda.zeebe.protocol.record.value.IncidentRecordValue;
-import io.camunda.zeebe.protocol.record.value.ProcessInstanceRecordValue;
 import io.camunda.zeebe.test.util.collection.Maps;
 import io.camunda.zeebe.test.util.record.RecordingExporter;
 import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.Collection;
 import java.util.function.Consumer;
 import org.junit.ClassRule;
 import org.junit.Rule;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
 
-public class ServiceTaskIncidentTest {
+@RunWith(Parameterized.class)
+public class ServiceTaskLikeIncidentTest {
 
   @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
 
   private static final String PROCESS_ID = "process";
-  private static final String SERVICE_TASK_ID = "task";
+  private static final String TASK_ELEMENT_ID = "task";
 
   @Rule
   public final RecordingExporterTestWatcher recordingExporterTestWatcher =
       new RecordingExporterTestWatcher();
 
-  private static BpmnModelInstance createBPMNModel(final Consumer<ServiceTaskBuilder> consumer) {
-    final var builder =
-        Bpmn.createExecutableProcess(PROCESS_ID).startEvent().serviceTask(SERVICE_TASK_ID);
+  @Parameter public TaskBuilder taskBuilder;
 
-    builder.zeebeJobType("test"); // default job type, can be overridden by consumer
+  @Parameters(name = "{0}")
+  public static Collection<Object[]> parameters() {
+    return ServiceTaskLikeBuilderProvider.buildersAsParameters();
+  }
```

Comment:
Should a similar construct be used in the `JobActivationIncidentTest` and `JobFailIncidentTest`? If so, I think it's fine to leave that out of scope of this pull request. But, let's create an issue for it then.

Reply:
Both test classes are focused on jobs. There are independent of the actual task. I don't think that we could cover more cases if we would morph them into parameterized tests.